ApiDoctor.Console/Program.cs

```diff
@@ -2562,6 +2557,210 @@ private static HttpRequest PreProcessUrlForSnippetGeneration(HttpRequest request
 
             return request;
         }
+        #endregion
+
+        #region generate permission files
+        /// <summary>
+        /// Generate/update permission files for operation topics
+        /// </summary>
+        /// <param name="options"></param>
+        /// <param name="issues"></param>
+        /// <returns></returns>
+        private static async Task<bool> GeneratePermissionFilesAsync(GeneratePermissionFilesOptions options, IssueLogger issues, DocSet docs = null)
+        {
+            // we don't care about validating documents in this context
+            options.IgnoreErrors = options.IgnoreWarnings = true;
+
+            var docSet = docs ?? await GetDocSetAsync(options, issues);
+            if (null == docSet)
+            {
+                return false;
+            }
+
+            // we only expect to have permission definitions in documents of ApiPageType
+            var docFiles = docSet.Files.Where(x => x.DocumentPageType == DocFile.PageType.ApiPageType);
+
+            // generate permissions document
+            var permissionsDocument = new PermissionsDocument();
+            if (docFiles.Any() && !options.BootstrappingOnly)
+            {
+                using var fileStream = new FileStream(options.PermissionsSourceFile, FileMode.Open);
```

Comment:
if the `PermissionsSourceFile` is a url, do we need to make a http request and pass the response stream?

Reply:
Sure.